### ISSUE TITLE
Using onReceiveProgress on "get" causes the error:  type 'List<String>' is not a subtype of type 'String'

### ISSUE THREAD

**Issue by kabnfever** [flutterchina/dio#515]
### New Issue Checklist

- [X ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info
```
| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | android                                               |      |
| Platform Version               | 9.0                                                   |      |
| Dio Version                    | 3.0.3                                                 |      |
| Android Studio / VS Code Versn | Android Studio 3.5 / VS Code 1.39.2                   |      |
| Repro rate                     | all the time (100%)                                   |      |
| Repro with our demo prj        | Yes, in example/download.dart                         |      |
| Demo project link              | https://github.com/flutterchina/dio/blob/master/example/download.dart |      |

**Flutter Doctor output:**
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, v1.9.1+hotfix.4, on Microsoft Windows [Version 10.0.18362.418], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
[√] Android Studio (version 3.5)
[√] VS Code, 64-bit edition (version 1.39.2)
[√] Connected device (3 available)

• No issues found!


```
### Issue Description and Steps
Whenever you use the "onReceiveProgress" option on a dio "get" you will receive the following error:
**DioError [DioErrorType.DEFAULT]: type 'List<String>' is not a subtype of type 'String'**

The following stack trace tells you exactly where the issue is and the solution is simple:
```
*** DioError ***:
uri: http://httpbin.org/
DioError [DioErrorType.DEFAULT]: type 'List<String>' is not a subtype of type 'String'
#0      DefaultTransformer.transformResponse (package:dio/src/transformer.dart:79)
<asynchronous suspension>
#1      DioMixin._dispatchRequest (package:dio/src/dio.dart:923)
<asynchronous suspension>
#2      DioMixin._request._interceptorWrapper.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:dio/src/dio.dart:828)
#3      DioMixin.checkIfNeedEnqueue (package:dio/src/dio.dart:1099)
#4      DioMixin._request._interceptorWrapper.<anonymous closure>.<anonymous closure> (package:dio/src/dio.dart:825)
#5      new Future..<anonymous closure> (dart:async/future.dart:176)
#6      _rootRun (dart:async/zone.dart:1120)
#7      _CustomZone.run (dart:async/zone.dart:1021)
#8      _CustomZone.runGuarded (dart:async/zone.dart:923)
#9      _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963)
#10     _rootRun (dart:async/zone.dart:1124)
#11     _CustomZone.run (dart:async/zone.dart:1021)
```

### Issue Resolution and Solution
The **transformResponse()** function in the **DefaultTransformer** class inside the **transformer.dart** source has an issue at line #79 which is the call to int.parse() in the following block:
```
    if (showDownloadProgress) {
      length = int.parse(response.headers[Headers.contentLengthHeader] ?? "-1");
    }
```
**response.headers** is declared as a ``Map<String, List<String>>`` and the code **response.headers[Headers.contentLengthHeader]** returns a ``List<String>`` NOT a ``String`` as the int.parse() requires, thus the reported error.

I have already created an override of the **transformResponse()** function with my correction noted below in my app and the transform now works just fine.  However, I know you want your code base to be correct, so I have included the code solution. The solution is to change the offending code block noted above to the following:
```
    if (showDownloadProgress) {
      length = int.parse(response.headers[Headers.contentLengthHeader].?elementAt(0) ?? "-1");
    }
```
Since there will only be one value for "content-length" you can always reference element zero of the value list of strings to get the "content-length" value as shown above. As I said, I already have this working and of course it incorporates the use of "-1" when the reply payload does not return a "content-length" header as json requests frequently do.

If you have any questions, feel free to reach out to me.

Regards.
Kevin Baroni

**Comment by wendux** [comment]
Fixed at v3.0.4

**Comment by kabnfever** [comment]
Thank you!  v3.0.4 works like a charm :).
